ext/DynamicQuantitiesLinearAlgebraExt.jl

```diff
@@ -1,11 +1,94 @@
 module DynamicQuantitiesLinearAlgebraExt
 
-using DynamicQuantities: UnionAbstractQuantity, QuantityArray, ustrip, dimension, new_quantity
+using DynamicQuantities: UnionAbstractQuantity, QuantityArray, ustrip, dimension, new_quantity, array_op
 
-import LinearAlgebra: norm
-import DynamicQuantities: _norm
+import LinearAlgebra: svd, Diagonal, eigen, det, diagm
 
-_norm(u::AbstractArray) = norm(u)
-norm(q::UnionAbstractQuantity, p::Real=2) = new_quantity(typeof(q), norm(ustrip(q), p), dimension(q))
+using LinearAlgebra: LinearAlgebra as LA, Algorithm, default_svd_alg, SVD, Adjoint, eigsortby, Eigen
+using DynamicQuantities:
+    DynamicQuantities as DQ, UnionAbstractQuantity, allequal, ustrip, dimension,
+    new_quantity, AbstractDimensions, QuantityArray,
+    constructorof, quantity_type
+using TestItems: @testitem
+
+DQ.is_ext_loaded(::Val{:LinearAlgebra}) = true
+DQ.norm(u) = LA.norm(u)
+LA.norm(q::UnionAbstractQuantity, p::Real=2) = new_quantity(typeof(q), LA.norm(ustrip(q), p), dimension(q))
+
+# Deal with ambiguous array operations:
+for op in (:(Base.:*), :(Base.:/), :(Base.:\)),
+    Q_ARRAY_TYPE in (:(QuantityArray{<:Any,1}), :(QuantityArray{<:Any,2})),
+    ARRAY_TYPE in (
+        LA.Transpose{<:Any, <:AbstractVector},
+        LA.Transpose{<:Any, <:AbstractMatrix},
+        LA.Transpose{<:Any, <:LA.Bidiagonal},
+        LA.Adjoint{<:Any, <:AbstractVector},
+        LA.Adjoint{<:Any, <:AbstractMatrix},
+        LA.Adjoint{<:Any, <:LA.Bidiagonal},
+        LA.Adjoint{<:Number, <:AbstractVector},
+        Union{LA.Transpose{T,V}, LA.Adjoint{T,V}} where {T,V<:AbstractVector},
+        Union{LA.Transpose{T,V}, LA.Adjoint{T,V}} where {T,V<:LA.Bidiagonal},
+        LA.AbstractTriangular,
+        Union{LA.LowerTriangular,LA.UpperTriangular},
+        Union{LA.UnitLowerTriangular,LA.UnitUpperTriangular},
+        LA.Diagonal,
+        LA.Bidiagonal,
+        LA.SymTridiagonal,
+        Union{LA.Hermitian{T,S}, LA.Symmetric{T,S}} where {T,S},
+    ),
```

Comment:
(Just added this after a lot of pain)

Surely there's a better way to deal with this sort of stuff for custom array types? Or is this what we have to do?

Reply:
(Leaving this blob of disambiguities to figure out later. Doesn't seem to affect the load time at all which is good. Just seems ugly and hard to maintain.)